apps/list.c

```diff
@@ -277,6 +277,57 @@ static void list_kdfs(void)
     sk_EVP_KDF_pop_free(kdfs, EVP_KDF_free);
 }
 
+/*
+ * RANDs
+ */
+DEFINE_STACK_OF(EVP_RAND)
+
+static int rand_cmp(const EVP_RAND * const *a, const EVP_RAND * const *b)
+{
+    int ret = strcasecmp(EVP_RAND_name(*a), EVP_RAND_name(*b));
+
+    if (ret == 0)
+        ret = strcmp(OSSL_PROVIDER_name(EVP_RAND_provider(*a)),
+                     OSSL_PROVIDER_name(EVP_RAND_provider(*b)));
+
+    return ret;
+}
+
+static void collect_rands(EVP_RAND *rand, void *stack)
+{
+    STACK_OF(EVP_RAND) *rand_stack = stack;
+
+    sk_EVP_RAND_push(rand_stack, rand);
+    EVP_RAND_up_ref(rand);
+}
+
+static void list_rands(void)
+{
+    STACK_OF(EVP_RAND) *rands = sk_EVP_RAND_new(rand_cmp);
+    int i;
+
+    BIO_printf(bio_out, "Provided RNGs and seed sources:\n");
+    EVP_RAND_do_all_provided(NULL, collect_rands, rands);
+    sk_EVP_RAND_sort(rands);
+    for (i = 0; i < sk_EVP_RAND_num(rands); i++) {
+        const EVP_RAND *m = sk_EVP_RAND_value(rands, i);
+
+        BIO_printf(bio_out, "  %s", EVP_RAND_name(m));
+        BIO_printf(bio_out, " @ %s\n",
```

Comment:
nit -- "from" (or similar0 rather than the atsign

Reply:
Other instances of similar use '@'.  I'd prefer _from_ but consistency is more important.

Reply:
This is new in master, so feel free to change it.  My only worry is what's more easily parsable by human eye or brain, "foo @ bar" or "foo from bar"?  To my eye, it's easier to distinguish the two halves apart with the atsign.

Reply:
BTW, the spaces around the atsign are there to make it look less like an email address...  said just in case you were going to argue with that :wink:

Reply:
This is a matter for a different PR.  Hopefully merged after this one since I've dealt with more than enough conflicts already -- albeit nothing like the number @levitte and @slontis have had to recently.